security/pkg/k8s/controller/workloadsecret.go

```diff
@@ -1,611 +0,0 @@
-	// Only used in self signed CA mode.
-	syncWithSelfSignedCaSecret bool
-}
-
-// NewSecretController returns a pointer to a newly constructed SecretController instance.
-func NewSecretController(ca certificateAuthority, enableNamespacesByDefault bool,
-	certTTL time.Duration, gracePeriodRatio float32, minGracePeriod time.Duration,
-	dualUse bool, core corev1.CoreV1Interface, forCA bool, pkcs8Key bool, namespaces []string,
-	dnsNames map[string]*DNSNameEntry, istioCaStorageNamespace, rootCertFile string,
-	selfSignedCa bool) (*SecretController, error) {
-
-	if gracePeriodRatio < 0 || gracePeriodRatio > 1 {
-		return nil, fmt.Errorf("grace period ratio %f should be within [0, 1]", gracePeriodRatio)
-	}
-	if gracePeriodRatio < recommendedMinGracePeriodRatio || gracePeriodRatio > recommendedMaxGracePeriodRatio {
-		k8sControllerLog.Warnf("grace period ratio %f is out of the recommended window [%.2f, %.2f]",
-			gracePeriodRatio, recommendedMinGracePeriodRatio, recommendedMaxGracePeriodRatio)
-	}
-
-	c := &SecretController{
-		ca:                         ca,
-		certTTL:                    certTTL,
-		istioCaStorageNamespace:    istioCaStorageNamespace,
-		gracePeriodRatio:           gracePeriodRatio,
-		certUtil:                   certutil.NewCertUtil(int(gracePeriodRatio * 100)),
-		caSecretController:         NewCaSecretController(core),
-		rootCertFile:               rootCertFile,
-		enableNamespacesByDefault:  enableNamespacesByDefault,
-		minGracePeriod:             minGracePeriod,
-		dualUse:                    dualUse,
-		core:                       core,
-		forCA:                      forCA,
-		pkcs8Key:                   pkcs8Key,
-		namespaces:                 make(map[string]struct{}),
-		dnsNames:                   dnsNames,
-		monitoring:                 newMonitoringMetrics(),
-		lastKCBSyncTime:            time.Time{},
-		syncWithSelfSignedCaSecret: selfSignedCa,
-	}
-
-	for _, ns := range namespaces {
-		c.namespaces[ns] = struct{}{}
-	}
-
-	// listen for service account creation across all listened namespaces, filter out which should be ignored upon receipt
-	saLW := listwatch.MultiNamespaceListerWatcher(namespaces, func(namespace string) cache.ListerWatcher {
-		return &cache.ListWatch{
-			ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
-				return core.ServiceAccounts(namespace).List(options)
-			},
-			WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
-				return core.ServiceAccounts(namespace).Watch(options)
-			},
-		}
-	})
-	c.saStore, c.saController =
-		cache.NewInformer(saLW, &v1.ServiceAccount{}, time.Minute, cache.ResourceEventHandlerFuncs{
-			AddFunc:    c.saAdded,
-			DeleteFunc: c.saDeleted,
-		})
-
-	istioSecretSelector := fields.SelectorFromSet(map[string]string{"type": IstioSecretType}).String()
-	scrtLW := listwatch.MultiNamespaceListerWatcher(namespaces, func(namespace string) cache.ListerWatcher {
```

Comment:
this part is the namespace selection.

(replacing/deleting this with istiod multi version is totally fine and better even to me. just want to ensure that i can track the progress and update the doc when it's ready, e.g. we have https://istio.io/pt-br/docs/ops/configuration/mesh/secret-creation/, and one of original use cases, https://github.com/istio/istio/issues/6182 )